pkgs/development/tools/language-servers/nixd/default.nix

```diff
@@ -129,6 +129,13 @@ in
         boost182
       ];
 
+      nativeBuildInputs = [
+        meson
+        ninja
+        pkg-config
+        llvmPackages.llvm
```

Comment:
I don't see the rationale? What's the error behind this?

Reply:
currently, `nix build .#pkgsCross.aarch64-multiplatform.nixd` fails to build because `llvmPackages.llvm` is a build time dependency of `nixd`. To fix, it needs to be added to `nativeBuildInputs`.

Reply:
> fails to build because

Sorry for my confusing words, I know you may submitting patches in order to improve nixd cross compilation, but how about pasting some logs here? For example, detailed error information about "missing" llvm native build inputs.

Reply:
Sorry for being confusing also! Here is the log: https://gist.github.com/danielbarter/862362de655b0ca7d130cd742684caa2

nothing really interesting going on, just the missing runtime dep

Reply:
> meson.build:34:7: ERROR: Dependency lookup for llvm with method 'cmake' failed: CMake binary for machine host machine not found. Giving up.

They how about adding "cmake" to `nativeBuildInputs`? Use cmake modules to find llvm then.

Reply:
adding `llvmPackages.llvm` to `nativeBuildInputs` is the correct solution here.

Reply:
> Why would we add cmake as a build time dependency to just tell meson how to find another build time dependency.

As per my understanding, meson have several ways to find llvm. The first choice is "llvm-config" binary tool, and the second one is the LLVM cmake module. LLVM itself is built from cmake thus it is make sense to add "cmake" to find it. "llvm-config" binary tool is suitable for finding llvm configured with suitable installing prefix and may not suitable for cross-compiling.

> another build time dependency.

I suppose if you have `cmake` in `nativeBuildInputs`, meson will not complain missing llvm as build time dep. LLVM supporting libraries are required as a runtime library, it is *indeed* runtime deps, not a build time dep.

Reply:
OK fine.

`nixd` does not cross compile. You guys fix it.